build.go

```diff
@@ -181,33 +182,49 @@ func (c *Client) validateImageReference(imageName string) (name.Reference, error
 	return ref, nil
 }
 
-func (c *Client) processAppDir(appDir string) (string, error) {
+func (c *Client) processAppPath(appPath string) (string, error) {
 	var (
-		resolvedAppDir = appDir
-		err            error
+		resolvedAppPath = appPath
+		err             error
 	)
 
-	if appDir == "" {
-		if appDir, err = os.Getwd(); err != nil {
-			return "", err
+	if appPath == "" {
+		if appPath, err = os.Getwd(); err != nil {
+			return "", errors.Wrap(err, "get working dir")
 		}
 	}
 
-	if resolvedAppDir, err = filepath.EvalSymlinks(appDir); err != nil {
-		return "", err
+	if resolvedAppPath, err = filepath.EvalSymlinks(appPath); err != nil {
+		return "", errors.Wrap(err, "evaluate symlink")
 	}
 
-	if resolvedAppDir, err = filepath.Abs(resolvedAppDir); err != nil {
-		return "", err
+	if resolvedAppPath, err = filepath.Abs(resolvedAppPath); err != nil {
+		return "", errors.Wrap(err, "resolve absolute path")
 	}
 
-	if fi, err := os.Stat(resolvedAppDir); err != nil {
-		return "", err
-	} else if !fi.IsDir() {
-		return "", fmt.Errorf("%s is not a directory", appDir)
+	fi, err := os.Stat(resolvedAppPath)
+	if err != nil {
+		return "", errors.Wrap(err, "stat file")
+	}
+
+	if !fi.IsDir() {
+		fh, err := os.Open(resolvedAppPath)
+		if err != nil {
+			return "", errors.Wrap(err, "read file")
+		}
+		defer fh.Close()
+
+		isZip, err := archive.IsZip(fh)
```

Comment:
@sclevine Decided to check early for a better user experience. We had talked offline about just letting the reader fail, but this will notify the user earlier (before pulling images, trying to start a build, etc).